pkg/api/validation/events.go

```diff
@@ -25,13 +29,35 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	// TODO: There is no namespace required for node.
-	if event.InvolvedObject.Kind != "Node" &&
-		event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
+
+	nonNamespacedKindFlag, err := isNonNamespacedKind(event.InvolvedObject.Kind, event.InvolvedObject.APIVersion)
+	if err != nil {
+		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.Kind", event.InvolvedObject.Kind, fmt.Sprintf("couldn't check whether namespace is allowed: %s", err)))
```

Comment:
The original ValidateEvent suppose all kinds except "Node" are namespaced. I updated this PR that supposes 3rd party objects from other groups are namespaced to accept  those events reference 3rd party as the original ValidateEvent. Is this acceptable?